src/libAtomVM/opcodesswitch.h

```diff
@@ -737,8 +754,9 @@ static void destroy_extended_registers(Context *ctx, unsigned int live)
 #define READ_DEST_REGISTER(dreg) *(dreg)
 
 #define READ_DEST_REGISTER_GC_SAFE(dreg_gc_safe) \
```

Comment:
When reading this file I wondered why most of the supporting code is inside macros. Do you need to always inline them for speed (like `__attribute__((always_inline))`) or you need them to be generic?

Reply:
as far as I remember from some tests I did lot of time ago, the compiler wasn't performing same level of optimization, but I need to check that.